tests/unit/helpers/format-message-test.js

```diff
@@ -179,7 +179,7 @@ test('able to discover all register translations', function(assert) {
   assert.expect(1);
   this.intl.addTranslation('es_MX', 'foo', 'bar'); /* tests that the locale name becomes normalized to es-mx */
   this.intl.exists('test', 'fr-ca');
-  assert.equal(this.intl.get('locales').join('; '), 'en-us; es-es; fr-fr; es-mx');
```

Comment:
would it be possible to split this PR up into multiple PRs? it has become pretty larger and hard to review... 😞 

e.g. the `@private` comments seem unrelated and the build-time locale list building could probably be implemented independent from the original changes in this PR, right?